### PR TITLE
Fixes parsing of new date format in Smart Insider transactions data

### PR DESCRIPTION
**Pull request by gsalaz98**
#### Description

* Fixes parsing of new date format in Smart Insider transactions data
* Makes converter throw hard error if it fails to parse from raw data
* Adds new unit tests
* Fixes typos where we called data `CSV` instead of `TSV`

#### Related Issue

N/A
#### Motivation and Context

Data integrity
#### Requires Documentation Change

N/A
#### How Has This Been Tested?

New unit tests + some reprocessing
#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [ ] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Data/Custom/SmartInsider/SmartInsiderTransaction.cs [hunk 1]
```diff
@@ -357,5 +357,25 @@ public override string ToLine()
...
+        /// </summary>
+        /// <param name="date">Date string to parse</param>
+        /// <returns>DateTime object</returns>
+        /// <exception cref="ArgumentException">Date string was unable to be parsed</exception>
+        public DateTime ParseTransactionDate(string date)
+        {
+            date = date.Replace(" ", "").Trim();
+
+            DateTime time;
+            if (!Parse.TryParseExact(date, "dd/MM/yyyyHH:mm:ss", DateTimeStyles.None, out time) &&
```
Comment (Jay-Jay-D) [inline Common/Data/Custom/SmartInsider/SmartInsiderTransaction.cs]: Lovely!

File: Common/Data/Custom/SmartInsider/SmartInsiderTransaction.cs [hunk 2]
```diff
@@ -357,5 +357,25 @@ public override string ToLine()
...
                 TreasuryHolding);
         }
+
+        /// <summary>
+        /// Attempts to normalize and parse SmartInsider transaction dates.
+        /// </summary>
+        /// <param name="date">Date string to parse</param>
+        /// <returns>DateTime object</returns>
+        /// <exception cref="ArgumentException">Date string was unable to be parsed</exception>
+        public DateTime ParseTransactionDate(string date)
```
Comment (Martin-Molinero) [inline Common/Data/Custom/SmartInsider/SmartInsiderTransaction.cs]: nit - maybe this guy could be static

Seems `SmartInsiderIntention` is performing similar operations, think we should address it there too https://github.com/QuantConnect/Lean/blob/master/Common/Data/Custom/SmartInsider/SmartInsiderIntention.cs#L157, maybe move this method into the `SmartInsiderEvent`

File: Tests/Common/Data/Custom/SmartInsiderTests.cs [hunk 3]
```diff
@@ -35,5 +36,26 @@ public void ErrorGetsMappedToSatisfyStockVesting()
             Assert.AreEqual(intention.ExecutionHolding, SmartInsiderExecutionHolding.SatisfyStockVesting);
             Assert.AreEqual(transaction.ExecutionHolding, SmartInsiderExecutionHolding.SatisfyStockVesting);
         }
+
+        [TestCase("2019-01-01  23:59:59")]
+        [TestCase("01/01/2019  23:59:59")]
+        public void ParsesOldAndNewTransactionDateTimeValues(string date)
```
Comment (Martin-Molinero) [inline Tests/Common/Data/Custom/SmartInsiderTests.cs]: Awesome tests! 👍

File: Common/Data/Custom/SmartInsider/SmartInsiderEvent.cs [hunk 4]
```diff
@@ -261,5 +261,25 @@ public SmartInsiderEvent(string tsvLine)
...
+        /// <exception cref="ArgumentException">Date string was unable to be parsed</exception>
+        public static DateTime ParseDate(string date)
+        {
+            date = date.Replace(" ", "");
+
+            DateTime time;
+            if (!Parse.TryParseExact(date, "yyyy-MM-ddHH:mm:ss", DateTimeStyles.None, out time) &&
+                !Parse.TryParseExact(date, "dd/MM/yyyyHH:mm:ss", DateTimeStyles.None, out time))
+            {
+                throw new ArgumentException($"SmartInsider transaction contains unparsable DateTime: {date}");
```
Comment (Martin-Molinero) [inline Common/Data/Custom/SmartInsider/SmartInsiderEvent.cs]: nit - I think exception message needs an update, to remove 'transaction', same for summary comment in line 266

**Pull request merged**